scripts/windows/InstallSecureBootKeys.ps1

```diff
@@ -0,0 +1,102 @@
+#Requires -RunAsAdministrator
+
+<#
+.DESCRIPTION
+This script installs and enables Secure Boot certificates at runtime (from the operating system).
+
+.PARAMETER PresignedObjectsPath
+A path to the directory containing the pre-signed Secure Boot objects.
+
+.PARAMETER PathToPkP7b
+(Optional) A path to the PKCS7 signature of the Platform Key (PK).
+
+.NOTES
+Author: Microsoft
+Date: 4/2/25
+Version: 1
+#>
+
+param(
+    [Parameter(Mandatory=$true)]
+    [string]$PresignedObjectsPath,
+
+    [Parameter(Mandatory=$false)]
+    [string]$PathToPkP7b
+)
+
+function Validate-FileSize {
+    param (
+        [string]$FilePath,
+        [int]$MinSizeBytes
+    )
+    if (-not (Test-Path -Path $FilePath)) {
+        throw "File not found: $FilePath"
+    }
+    $fileInfo = Get-Item -Path $FilePath
+    if ($fileInfo.Length -lt $MinSizeBytes) {
+        throw "File $FilePath is smaller than the required minimum size of $MinSizeBytes bytes."
+    }
+}
+
+function Log-Message {
+    param (
+        [string]$Message,
+        [string]$Color = "White"
+    )
+    Write-Host $Message -ForegroundColor $Color
+}
+
+try {
+    # Check if Secure Boot is already enabled
+    $SecurebootEnabled = Confirm-SecureBootUEFI
+    if ($SecurebootEnabled) {
+        Log-Message "Secure Boot is already enabled. Please disable Secure Boot and try again." "Red"
+        return
+    }
+
+    Log-Message "Enrolling certificates for Secure Boot..." "Green"
+
+    # Validate the presence of required pre-signed objects
+    $RequiredFiles = @("DefaultPk.bin", "DefaultKek.bin", "DefaultDb.bin", "DefaultDbx.bin")
+    foreach ($RequiredFile in $RequiredFiles) {
+        $FilePath = Join-Path -Path $PresignedObjectsPath -ChildPath $RequiredFile
+        if (-not (Test-Path -Path $FilePath)) {
+            Log-Message "Required file $RequiredFile is missing. Please download the latest release and try again." "Yellow"
+            Log-Message "https://github.com/microsoft/secureboot_objects/releases" "Yellow"
+            return
+        }
+    }
+
+    # Validate the size of the DBX file (minimum 28 bytes - EFI_SIGNATURE_LIST minimum size assuming 0 entries)
+    $DbxFilePath = Join-Path -Path $PresignedObjectsPath -ChildPath "DefaultDbx.bin"
+    Validate-FileSize -FilePath $DbxFilePath -MinSizeBytes 28
+
+    # Timestamp for Secure Boot enrollment
+    $time = "2015-08-28T00:00:00Z"
+
+    # Enroll certificates in reverse order
```

Comment:
Would recommend checking output of Set-SecureBootUEFI to prevent the system from ending up in a torn state and being unbootable.